### DISCUSSION TITLE
I can't create Mozilla's Hub Cloud stack any more

### DISCUSSION THREAD

**Discussion by FrankAllanRJ** [mozilla/hubs#5826] [Hubs Cloud Questions]
**Description**
Hi. 
I'm trying to create a Mozilla's Hubs Cloud stack, but I always getting "Template error"

Status reason: "Template error: Unable to get mapping for Regions::us-east-1::Abbreviation. Rollback requested by user."

How to solve it?

**To Reproduce**
1. AWS Marketplace -> Hubs Cloud Enterprise -> Launch CloudFormation (US East (N. Virginia) -> Launch (1 - Cloud Enterprise hubs.jpg)

2. AWS CloudFormation Create stack: keeping an entered S3 URL with template (https://s3.amazonaws.com/awsmp-fulfillment-cf-templates-prod/94ec4ef1-285d-4e63-b2eb-0b65ea93b942/a81613f05c734088b94ae017ddc2b739.template). Next (2 - Hubs Cloud Enterprise-CloudFormation Stack.jpg).

3. The error appears already in the 3rd stack creation event (3 -metav-error.jpg)

**Expected behavior**
I expected the stack to be created successfully. But it always fails.

**Screenshots**
Here are the screenshots:
![1 - hubs Cloud Enterprise](https://user-images.githubusercontent.com/114261183/204303326-25bd96cd-5083-48fc-b946-e17b39488197.jpg)
![2 - hubs Cloud Enterprise-CloudFormation Stack](https://user-images.githubusercontent.com/114261183/204303317-b8a5008a-4520-477f-ba81-2bdaea171b65.jpg)
![3 - metadev-error](https://user-images.githubusercontent.com/114261183/204303330-e34efef3-ecfa-4ffc-9858-b301a2666628.jpg)

**Additional context**
I'm trying to create a Hubs Cloud version 1.1.5 stack, but I've already tried to create a version 1.1.3 and 1.1.2 and it didn't work either. The errors are different in older versions (it seems to have to do with the database version).
I noticed that version 1.1.4, which was the one I was able to create stacks normally, disappeared.

In the attachment below, AWS support report on their analysis of the problem:
[Hubs Cloud Enterprise-AWSCloudFormation Stack error.pdf](https://github.com/mozilla/hubs/files/10104634/Hubs.Cloud.Enterprise-AWSCloudFormation.Stack.error.pdf)

**Comment by pattersonbl2** [comment]
I have looked at the error and the information sent to you from AWS.  I am a little confused because my template mapping is in this order. 
`mappings:
  Regions:
    us-east-1:
      ImageId: 
    us-east-2:
      ImageId: 
    us-west-2:
      ImageId: 
    eu-west-1:
      ImageId: 
    ap-northeast-1:
      ImageId: `
I am not seeing an abbreviation in mapping.  This template is hosted and deployed by aws, Which makes the support answer very confusing.
I have just deployed an instance with same template version  1.1.5 without any issues.  i also didn't make any changes to the autopopulated template. 

1.1.4 was removed due to an error by the aws team.  I had requested it be added back but they were unable to do. 


Was this your first error  ?

**Comment by FrankAllanRJ** [comment]
Yes, @Pattersonbl2. 
Yup. This was the only error I got when trying to create a stack in version 1.1.5.
I opened the template file (attached in the original post) in Notepad++ and searched for the word "Abbreviation" and found this: ![image](https://user-images.githubusercontent.com/114261183/204659107-949cb75a-6dbb-4f0e-ad97-292f55312510.png)

What do you recommend I do in this case? Replace word ("Abbreviation") with another value? Erase it?

Best, 

Frank Allan.

**Reply by pattersonbl2** [reply]
You can try it.  I didn't remove it and it worked like normal for me.

**Reply by FrankAllanRJ** [reply]
Hello Petterson.

Reviewing my conversation with the AWS support team, I noticed that they actually mention "error" on the presence of the Abbreviation key in the Mappings section. However, comparing the 2 templates I used (1.1.5 and 1.1.4), I noticed that in version 1.1.4, the key exists in all mentioned regions. Already in version 1.1.5, no.

v1.1.4 (https://s3.amazonaws.com/awsmp-fulfillment-cf-templates-prod/94ec4ef1-285d-4e63-b2eb-0b65ea93b942.d0ef2f97-f5c6-4c1e-bf49-9bdc24320a12.template)
	Mappings:
	  Regions:
		us-east-1:
-->		  Abbreviation: USE1
		  ImageId: ami-03405d39fd9908f1f
		us-west-2:
-->		  Abbreviation: USW2
		  ImageId: ami-0672dfbd435e4214a
		us-east-2:
-->		  Abbreviation: USE2
		  ImageId: ami-0f8217f898c479450
		ap-northeast-1:
-->		  Abbreviation: APN1
		  ImageId: ami-0fb6a019bc3c87aa4
		eu-west-1:
-->		  Abbreviation: EU
		  ImageId: ami-0dd148167284b27c8


v1.1.5 (https://s3.amazonaws.com/awsmp-fulfillment-cf-templates-prod/94ec4ef1-285d-4e63-b2eb-0b65ea93b942/a81613f05c734088b94ae017ddc2b739.template):
Mappings:
  Regions:
    us-east-1:
      ImageId: ami-0c9f1e50422b0f1f3
    us-east-2:
      ImageId: ami-0cb9e388814d9aff0
    us-west-2:
      ImageId: ami-0a4d84a06a75aa640
    eu-west-1:
      ImageId: ami-079b9004fd6f90bf3
    ap-northeast-1:
      ImageId: ami-05f94a40e284ff441

I'm assuming the error is the absence of the Abbreviation keys in each of the regions.
I'm going to test the new template (1.1.5) by including the Abbreviation keys in the Mappings section. However, even if it works, I am interested in making it work with the template hosted by AWS, so that my team and I are not dependent on custom templates.

When you reported that you were able to create a stack using the 1.1.5 template, was it on AWS? If so, I'll have to pass on to you exactly all the information I'm using to create my stack, so you can analyze whether I'm making any mistakes. But, at the same time, I think it's strange that I'm doing something wrong, because with the old template (1.1.4), I can create the stack without any problem.

Best,

Frank Allan

**Reply by pattersonbl2** [reply]
When I deployed my stack it was using the 1.1.5 version.  This didn't have the abbreviation keys and still worked.   I didn't make any modifications to the template.  This template is the version hosted by aws.  I have had our QA team test this same version and they found no issues.   

You should never make any modifications because these template come from a S3 bucket hosted by aws.  This will overwrite any changes you make because it "should" be pulling the hosted template from that S3 bucket URL every time you create a stack.  If you taking this template from the URL then putting them in your own s3 bucket, this would be incorrect as well because it would be referencing resources not in your account yet.   This would make sense why 1.1.4 is working and 1.1.5 isn't.   In your previous message, You can see how the Ami's are different.  During the creation process it would reference each item and import them into your account if not present.

**Reply by pattersonbl2** [reply]
I am going to run a test today with all the url given to see if i can build a stack with them.

**Reply by FrankAllanRJ** [reply]
I performed the test I promised in the previous comment (testing the new template (1.1.5) including the abbreviation keys in the Mappings section) and it worked fine.
![image](https://user-images.githubusercontent.com/114261183/206035311-e139a7d3-3f9d-4265-93fb-59d531403317.png)

I hosted it in my bucket. Follows: ([https://iptv-backups.s3.us-east-2.amazonaws.com/templates/a81613f05c734088b94ae017ddc2b739-1.1.5-fixit.template])

I understand your concern and agree that it is best not to use custom templates. And I noticed that the Ami's are different, but it worked.

I tried to create a stack again with the 1.1.5 template without my modifications and it didn't work.

I don't know what I'm doing wrong using version 1.1.5 without modifications. But at the same time, I think I'm doing everything right, as I can create with the 1.1.4 templates (unmodified) and with the modified 1.1.5 version.

Best, 

Frank

**Reply by pattersonbl2** [reply]
I am not sure. I can use the URL you sent first without any issues or modifications. The AMIs must be publicly available with any AWS account. This would make sense because if I use an ami in one account, it can be shared to any account I want.  If this is the case hosting the template yourself isn't a problem.  You would just have to remember if we update the template to do the same thing.  I am not sure if it's a region issue or an account issue at this point.  I know it may not be a bug but maybe something happening on the AWS side of things.

**Comment by FrankAllanRJ** [comment]
Hi Petterson. 

Did you understand why I can create stack in version 1.1.5 with just the abbreviation included?

**Comment by pattersonbl2** [comment]
I am stomped as to why this is the case.  AWS wasn't sure as well but they haven't been able to run a test to see if they can reproduce the issue.

**Comment by Andy-Roger-Kats** [comment]
@Pattersonbl2 I am running into the same issue as @FrankAllanRJ and would prefer to not modify the template if possible. The  screenshot below is from the confirmation screen just before the actual template launch. Curious if you have any other resolutions. I am attempting to update the stack to 1.1.5 as well. 

![Screen Shot 2022-12-15 at 8 51 13 AM](https://user-images.githubusercontent.com/56032062/207920106-2ee355b9-44f5-4c54-ad74-c18f9a4550b8.png)
